### ISSUE TITLE
Mitigation failures since upgrading from v0.6.3

### ISSUE THREAD

**Issue by RichardDavies** [PomDev2/wlanpoke#2]
_I'm migrating this issue from the SB forum so we can perform further troubleshooting and diagnostics_

I want to add another report of the mitigation not working in every case. I have four SB Radios connected to WiFi in my house. All four were successfully using wlanpoke v0.6.3 to mitigate all lost connections. I updated one of them to v0.7.0 (and then later to v0.7.1 and v0.7.2) and since updating it to 0.7.x it now loses its connection after about 12-24 hours and requires a manual reboot. When I updated it from 0.6.3 to 0.7.2, the only thing I changed with the configuration was to add "-W quick" to enable the quick webserver.

The other radios still using 0.6.3 are still staying connected without any issue. Can you think of any reason why 0.7.2 would be less reliable than 0.6.3 at restoring the lost WiFi connection? 

**Initial response from PomDev:**

> Sure, bugs or other problems. This is the second report of a failure to keep the radios always working.
> 
> Try saving and uploading the radio's logs, which are lost when located in /var/log. Make a '/etc/log' directory, which would survive a reboot, and use the -d /etc/log/ option to save logs to there, then upload, zip, and upload to here the relevant portion for examination.
> 
> It might be that the quick reset is causing trouble. Try setting the quick reset to 7 (higher than 6) to disable quick reset.
> 
> Try disabling the web server, and/or running the 'slow' version (-W slow option).
> 
> Swap the version 0.7.x radio with a 0.6.3 radio. Does the problem follow the 0.7.x version? Is there anything different about this radio (e.g., a different firmware version)?
> 
> Other troubleshooting involves moving the radio next to your desktop, and connecting and opening a terminal TTL serial connection (115200 bps) to the radio and launching wlanpoke from there. This would show all messages from boot-up to linux boot, etc. Save the terminal session and look for issues. If not obvious, you can increase the debug level for more verbosity. If that is sill not enough, we can add more debug statements to pinpoint what is not working.
> 
> I would do this here, but my 7 radios have been perfect, so far. However, my environment has significantly improved over the last few weeks, so this is not a good test site. That might be a clue.
> 
> A few additional questions:
> Does the radio respond to pings from the desktop? How about ssh? The optional web server?
> What does the squeezebox say under Network Info? Wireless Info?
> Can it play any music source? Is the wireless icon red?
> Does your router report the radio on some other ip address (look for its mac address)
> 
> I ask because the radio next to my desktop, connected via a TTL serial interface to a terminal, has kept playing (very softly) for days and days, and the wireless icon stays white. However, when I tried the QuickReset method using the terminal ("wpa_cli reassoc"), then looked at Settings | Advanced | Diagnostics | Network Health | Check Network returns with ip address "already in use", the Server Info section displays "Can't Ping" either mysqueezebox.com, and the LMS Library machine, yet ping succeeds from the serial terminal. And the music has not stopped, either from Pandora or the LMS library. There was also a "Wireless Info" display showing "Wireless network" "Not Connected", and so forth. It may be that the quick reset is somehow incomplete, causing confusion with the jive player app. I tried killing and relaunching jive, but the radio rebooted!
> 
> We don't want to confuse jive. If your radio stops playing music, for that radio, disable the quick reset and advance the full reset to the 3rd ping failure to speed things up. In the meantime, I'll look into this further.

**Comment by RichardDavies** [comment]
I downgraded back to v0.6.3 just to confirm I wasn't crazy. (I wasn't. The radio's connection has been solid again for 48+ hrs since downgrading.) I'll upgrade to the latest v0.7.x and try some of your troubleshooting steps report back soon with the results and diagnostic details.

**Comment by RichardDavies** [comment]
v0.7.3 with the webserver disabled still failed after about 12 hours. The Wireless Info reported "Not connected" and it had the red wireless icon. I had a SSH shell open, and here's the output it showed over the last few hours before it died:

> Selected interface 'eth1'
> root: wlan started
> Restarting dhcp
> udhcpc (v1.18.2) started
> killall: zcip: no process killed
> Sending discover...
> Performing a DHCP renew
> Sending discover...
> Sending select for 192.168.0.87...
> Lease of 192.168.0.87 obtained, lease time 86400
> killall: zcip: no process killed
> route: SIOCDELRT: No such process
> adding dns 1.1.1.1
> adding dns 192.168.0.1
> 2021-03-23T23:41:52-0700 Full: waiting for successful ping...
> 2021-03-24T00:00:01-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T00:00:01-0700 Quick: waiting for successful ping...
> 2021-03-24T00:00:20-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T00:00:20-0700 Quick: waiting for successful ping...
> 2021-03-24T00:00:33-0700 Stopping and restarting wlan...
> root: wlan: stopping
> ..unloading all
> killall: recEvent: no process killed
> root: wlan stopped
> root: wlan: starting
> Setting eth1 macaddr: 00:04:20:26:8E:09
> Load Board Data from /lib/atheros/calData_ar6102_15dBm.bin
> Updating MAC address
> BMI Write Memory (address: 0x502400, filename: /lib/atheros/athwlan.bin)
> BMI Write Memory (address: 0x52d8bc, filename: /lib/atheros/data.patch.hw2_0.bin)
> BMI Write Memory (address: 0x500418, value: 0x52d8bc)
> BMI Bit-Wise (OR) modify Register (address: 0x500410, orig:0x0, new: 0x1,  mask:0x1)
> BMI Done
> Selected interface 'eth1'
> root: wlan started
> Restarting dhcp
> udhcpc (v1.18.2) started
> killall: zcip: no process killed
> Sending discover...
> Performing a DHCP renew
> Sending discover...
> Sending select for 192.168.0.87...
> Lease of 192.168.0.87 obtained, lease time 86400
> killall: zcip: no process killed
> route: SIOCDELRT: No such process
> adding dns 1.1.1.1
> adding dns 192.168.0.1
> 2021-03-24T00:00:50-0700 Full: waiting for successful ping...
> 2021-03-24T00:25:30-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T00:25:30-0700 Quick: waiting for successful ping...
> 2021-03-24T00:25:42-0700 Stopping and restarting wlan...
> root: wlan: stopping
> ..unloading all
> killall: recEvent: no process killed
> root: wlan stopped
> root: wlan: starting
> Setting eth1 macaddr: 00:04:20:26:8E:09
> Load Board Data from /lib/atheros/calData_ar6102_15dBm.bin
> Updating MAC address
> BMI Write Memory (address: 0x502400, filename: /lib/atheros/athwlan.bin)
> BMI Write Memory (address: 0x52d8bc, filename: /lib/atheros/data.patch.hw2_0.bin)
> BMI Write Memory (address: 0x500418, value: 0x52d8bc)
> BMI Bit-Wise (OR) modify Register (address: 0x500410, orig:0x0, new: 0x1,  mask:0x1)
> BMI Done
> Selected interface 'eth1'
> root: wlan started
> Restarting dhcp
> udhcpc (v1.18.2) started
> killall: zcip: no process killed
> Sending discover...
> Performing a DHCP renew
> Sending discover...
> Sending select for 192.168.0.87...
> Lease of 192.168.0.87 obtained, lease time 86400
> killall: zcip: no process killed
> route: SIOCDELRT: No such process
> adding dns 1.1.1.1
> adding dns 192.168.0.1
> 2021-03-24T00:25:59-0700 Full: waiting for successful ping...
> 2021-03-24T00:42:31-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T00:42:31-0700 Quick: waiting for successful ping...
> 2021-03-24T00:42:55-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T00:42:55-0700 Quick: waiting for successful ping...
> 2021-03-24T00:43:53-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T00:43:53-0700 Quick: waiting for successful ping...
> 2021-03-24T00:44:15-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T00:44:15-0700 Quick: waiting for successful ping...
> 2021-03-24T00:44:27-0700 Stopping and restarting wlan...
> root: wlan: stopping
> ..unloading all
> killall: recEvent: no process killed
> root: wlan stopped
> root: wlan: starting
> Setting eth1 macaddr: 00:04:20:26:8E:09
> Load Board Data from /lib/atheros/calData_ar6102_15dBm.bin
> Updating MAC address
> BMI Write Memory (address: 0x502400, filename: /lib/atheros/athwlan.bin)
> BMI Write Memory (address: 0x52d8bc, filename: /lib/atheros/data.patch.hw2_0.bin)
> BMI Write Memory (address: 0x500418, value: 0x52d8bc)
> BMI Bit-Wise (OR) modify Register (address: 0x500410, orig:0x0, new: 0x1,  mask:0x1)
> BMI Done
> Selected interface 'eth1'
> root: wlan started
> Restarting dhcp
> udhcpc (v1.18.2) started
> killall: zcip: no process killed
> Sending discover...
> Performing a DHCP renew
> Sending discover...
> Sending select for 192.168.0.87...
> Lease of 192.168.0.87 obtained, lease time 86400
> killall: zcip: no process killed
> route: SIOCDELRT: No such process
> adding dns 1.1.1.1
> adding dns 192.168.0.1
> 2021-03-24T00:44:45-0700 Full: waiting for successful ping...
> 2021-03-24T01:00:14-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T01:00:15-0700 Quick: waiting for successful ping...
> 2021-03-24T01:00:27-0700 Stopping and restarting wlan...
> root: wlan: stopping
> ..unloading all
> killall: recEvent: no process killed
> root: wlan stopped
> root: wlan: starting
> Setting eth1 macaddr: 00:04:20:26:8E:09
> Load Board Data from /lib/atheros/calData_ar6102_15dBm.bin
> Updating MAC address
> BMI Write Memory (address: 0x502400, filename: /lib/atheros/athwlan.bin)
> BMI Write Memory (address: 0x52d8bc, filename: /lib/atheros/data.patch.hw2_0.bin)
> BMI Write Memory (address: 0x500418, value: 0x52d8bc)
> BMI Bit-Wise (OR) modify Register (address: 0x500410, orig:0x0, new: 0x1,  mask:0x1)
> BMI Done
> Selected interface 'eth1'
> root: wlan started
> Restarting dhcp
> udhcpc (v1.18.2) started
> killall: zcip: no process killed
> Sending discover...
> Sending discover...
> Sending discover...
> killall: zcip: no process killed
> No lease, forking to background
> 2021-03-24T01:00:51-0700 Full: waiting for successful ping...
> 2021-03-24T01:17:26-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T01:17:26-0700 Quick: waiting for successful ping...
> 2021-03-24T01:17:45-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T01:17:46-0700 Quick: waiting for successful ping...
> 2021-03-24T01:17:58-0700 Stopping and restarting wlan...
> root: wlan: stopping
> ..unloading all
> killall: recEvent: no process killed
> root: wlan stopped
> root: wlan: starting
> Setting eth1 macaddr: 00:04:20:26:8E:09
> Load Board Data from /lib/atheros/calData_ar6102_15dBm.bin
> Updating MAC address
> BMI Write Memory (address: 0x502400, filename: /lib/atheros/athwlan.bin)
> BMI Write Memory (address: 0x52d8bc, filename: /lib/atheros/data.patch.hw2_0.bin)
> BMI Write Memory (address: 0x500418, value: 0x52d8bc)
> BMI Bit-Wise (OR) modify Register (address: 0x500410, orig:0x0, new: 0x1,  mask:0x1)
> BMI Done
> Selected interface 'eth1'
> root: wlan started
> Restarting dhcp
> udhcpc (v1.18.2) started
> killall: zcip: no process killed
> Sending discover...
> Performing a DHCP renew
> Sending discover...
> Sending select for 192.168.0.87...
> Lease of 192.168.0.87 obtained, lease time 86400
> killall: zcip: no process killed
> route: SIOCDELRT: No such process
> adding dns 1.1.1.1
> adding dns 192.168.0.1
> 2021-03-24T01:18:15-0700 Full: waiting for successful ping...
> 2021-03-24T01:33:18-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T01:33:18-0700 Quick: waiting for successful ping...
> 2021-03-24T01:35:14-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T01:35:14-0700 Quick: waiting for successful ping...
> 2021-03-24T01:35:33-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T01:35:33-0700 Quick: waiting for successful ping...
> 2021-03-24T01:36:22-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T01:36:22-0700 Quick: waiting for successful ping...
> 2021-03-24T01:36:54-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T01:36:54-0700 Quick: waiting for successful ping...
> 2021-03-24T01:37:06-0700 Stopping and restarting wlan...
> root: wlan: stopping
> ..unloading all
> killall: recEvent: no process killed
> root: wlan stopped
> root: wlan: starting
> Setting eth1 macaddr: 00:04:20:26:8E:09
> Load Board Data from /lib/atheros/calData_ar6102_15dBm.bin
> Updating MAC address
> BMI Write Memory (address: 0x502400, filename: /lib/atheros/athwlan.bin)
> BMI Write Memory (address: 0x52d8bc, filename: /lib/atheros/data.patch.hw2_0.bin)
> BMI Write Memory (address: 0x500418, value: 0x52d8bc)
> BMI Bit-Wise (OR) modify Register (address: 0x500410, orig:0x0, new: 0x1,  mask:0x1)
> BMI Done
> Selected interface 'eth1'
> root: wlan started
> Restarting dhcp
> udhcpc (v1.18.2) started
> killall: zcip: no process killed
> Sending discover...
> Performing a DHCP renew
> Sending discover...
> Sending select for 192.168.0.87...
> Lease of 192.168.0.87 obtained, lease time 86400
> killall: zcip: no process killed
> route: SIOCDELRT: No such process
> adding dns 1.1.1.1
> adding dns 192.168.0.1
> 2021-03-24T01:37:23-0700 Full: waiting for successful ping...
> 2021-03-24T01:56:00-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T01:56:00-0700 Quick: waiting for successful ping...
> 2021-03-24T01:57:04-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T01:57:04-0700 Quick: waiting for successful ping...
> 2021-03-24T01:57:28-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T01:57:28-0700 Quick: waiting for successful ping...
> ping: sendto: Network is unreachable
> ping: sendto: Network is unreachable
> 2021-03-24T01:57:38-0700 Stopping and restarting wlan...
> root: wlan: stopping
> ..unloading all
> killall: recEvent: no process killed
> root: wlan stopped
> root: wlan: starting
> Setting eth1 macaddr: 00:04:20:26:8E:09
> Load Board Data from /lib/atheros/calData_ar6102_15dBm.bin
> Updating MAC address
> BMI Write Memory (address: 0x502400, filename: /lib/atheros/athwlan.bin)
> BMI Write Memory (address: 0x52d8bc, filename: /lib/atheros/data.patch.hw2_0.bin)
> BMI Write Memory (address: 0x500418, value: 0x52d8bc)
> BMI Bit-Wise (OR) modify Register (address: 0x500410, orig:0x0, new: 0x1,  mask:0x1)
> BMI Done
> Selected interface 'eth1'
> root: wlan started
> Restarting dhcp
> udhcpc (v1.18.2) started
> killall: zcip: no process killed
> Sending discover...
> Performing a DHCP renew
> Sending discover...
> Sending select for 192.168.0.87...
> Lease of 192.168.0.87 obtained, lease time 86400
> killall: zcip: no process killed
> route: SIOCDELRT: No such process
> adding dns 1.1.1.1
> adding dns 192.168.0.1
> 2021-03-24T01:57:56-0700 Full: waiting for successful ping...
> 2021-03-24T02:19:49-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T02:19:49-0700 Quick: waiting for successful ping...
> 2021-03-24T02:20:48-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T02:20:49-0700 Quick: waiting for successful ping...
> 2021-03-24T02:21:18-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T02:21:18-0700 Quick: waiting for successful ping...
> 2021-03-24T02:22:22-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T02:22:22-0700 Quick: waiting for successful ping...
> 2021-03-24T02:23:19-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T02:23:19-0700 Quick: waiting for successful ping...
> 2021-03-24T02:23:32-0700 Stopping and restarting wlan...
> root: wlan: stopping
> ..unloading all
> killall: recEvent: no process killed
> root: wlan stopped
> root: wlan: starting
> Setting eth1 macaddr: 00:04:20:26:8E:09
> Load Board Data from /lib/atheros/calData_ar6102_15dBm.bin
> Updating MAC address
> BMI Write Memory (address: 0x502400, filename: /lib/atheros/athwlan.bin)
> BMI Write Memory (address: 0x52d8bc, filename: /lib/atheros/data.patch.hw2_0.bin)
> BMI Write Memory (address: 0x500418, value: 0x52d8bc)
> BMI Bit-Wise (OR) modify Register (address: 0x500410, orig:0x0, new: 0x1,  mask:0x1)
> BMI Done
> Selected interface 'eth1'
> root: wlan started
> Restarting dhcp
> udhcpc (v1.18.2) started
> killall: zcip: no process killed
> Sending discover...
> Performing a DHCP renew
> Sending discover...
> Sending select for 192.168.0.87...
> Lease of 192.168.0.87 obtained, lease time 86400
> killall: zcip: no process killed
> route: SIOCDELRT: No such process
> adding dns 1.1.1.1
> adding dns 192.168.0.1
> 2021-03-24T02:23:49-0700 Full: waiting for successful ping...
> 2021-03-24T02:53:47-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T02:53:47-0700 Quick: waiting for successful ping...
> 2021-03-24T02:54:28-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T02:54:28-0700 Quick: waiting for successful ping...
> 2021-03-24T02:57:29-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T02:57:29-0700 Quick: waiting for successful ping...
> 2021-03-24T02:57:51-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T02:57:51-0700 Quick: waiting for successful ping...
> 2021-03-24T02:58:10-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T02:58:10-0700 Quick: waiting for successful ping...
> 2021-03-24T02:58:30-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T02:58:30-0700 Quick: waiting for successful ping...
> 2021-03-24T02:58:42-0700 Stopping and restarting wlan...
> root: wlan: stopping
> ..unloading all
> killall: recEvent: no process killed
> root: wlan stopped
> root: wlan: starting
> Setting eth1 macaddr: 00:04:20:26:8E:09
> Load Board Data from /lib/atheros/calData_ar6102_15dBm.bin
> Updating MAC address
> BMI Write Memory (address: 0x502400, filename: /lib/atheros/athwlan.bin)
> BMI Write Memory (address: 0x52d8bc, filename: /lib/atheros/data.patch.hw2_0.bin)
> BMI Write Memory (address: 0x500418, value: 0x52d8bc)
> BMI Bit-Wise (OR) modify Register (address: 0x500410, orig:0x0, new: 0x1,  mask:0x1)
> BMI Done
> Selected interface 'eth1'
> root: wlan started
> Restarting dhcp
> udhcpc (v1.18.2) started
> killall: zcip: no process killed
> Sending discover...
> Performing a DHCP renew
> Sending discover...
> Sending select for 192.168.0.87...
> Lease of 192.168.0.87 obtained, lease time 86400
> killall: zcip: no process killed
> route: SIOCDELRT: No such process
> adding dns 1.1.1.1
> adding dns 192.168.0.1
> 2021-03-24T02:58:59-0700 Full: waiting for successful ping...
> 2021-03-24T03:26:40-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T03:26:40-0700 Quick: waiting for successful ping...
> 2021-03-24T03:28:05-0700 Resetting wlan...
> Selected interface 'eth1'
> OK
> 2021-03-24T03:28:05-0700 Quick: waiting for successful ping...
> 2021-03-24T03:28:34-0700 Resetting wlan...

I will try logging to /etc/log and report back with the logs.

**Comment by PomDev2** [comment]
You can enable the 'slow' web server, it is apparently not involved.

It is looking like the issue is the wpa_cli process exiting and leaving the network stack without a way to handle certain networking messages. When that happens, a re-association from the 0.7.3 ResetQuick function is not followed by a request to renew the lease. The 0.7.4 version in the development branch adds a call to renew the lease, which seems to work very well. But the larger issue is what is happening to the wpa_cli process? This process is killed and restarted by the full reset, but it seems to disappear on its own when just using the quick reset. Even though my environment has quieted down, wpa_cli was not running on 4 out of 7 machines here.

The 0.7.5 software today has a new function wpa_cli_check that periodically checks the process, and restarts it if missing. You can test a simplified version of this from the ssh by entering (copy and paste):
  if ! pidof wpa_cli ; then echo "not running" ; /usr/sbin/wpa_cli -B -a/etc/network/wpa_action ; else echo "running" ; fi

The 0.7.4 version should work. You can try this until the 0.7.5 version is uploaded perhaps later on today.

I will try to move the existing debugging posts to the community build firmware topic away from the lost connection topic.

**Comment by RichardDavies** [comment]
> The wlanpoke 0.7.5 script is available in the GitHub development branch. Please use this as your baseline. If it is still unreliable "off the shelf," disable the quick reset method, and try again. If there is a problem even with it disabled, we need to know before this is merged into the main branch and into a release. 

Okay. I've just installed the development branch and will let you know if I continue to experience lost connections with it.

**Comment by RichardDavies** [comment]
Well, I thought maybe it was going to work with v0.7.5 running `/etc/wlanpoke/wlanpoke.sh -d /etc/log/ -W quick` because the connection had been stable for almost 48 hours, but unfortunately it lost its connection again last night. Here are [the logs from /etc/log](https://www.dropbox.com/s/1vy916lyc3fu3tg/logs.tar.gz?dl=0) and here is the most recent command line output before it lost the connection:

> nc: can't connect to remote host (192.168.0.5): Connection refused
> nc: can't connect to remote host (192.168.0.5): Connection refused
> nc: can't connect to remote host (192.168.0.5): Connection refused
> nc: can't connect to remote host (192.168.0.5): Connection refused
> nc: can't connect to remote host (192.168.0.5): Connection refused
> nc: can't connect to remote host (192.168.0.5): Connection refused
> 2021-03-26T00:00:21-0700 Resetting wlan... eth1
> Selected interface 'eth1'
> OK
> 2021-03-26T00:00:21-0700 Quick: waiting for successful ping...
> nc: can't connect to remote host (192.168.0.5): Connection refused
> 2021-03-26T00:00:39-0700 Resetting wlan... eth1
> Selected interface 'eth1'
> OK
> 2021-03-26T00:00:39-0700 Quick: waiting for successful ping...
> nc: can't connect to remote host (192.168.0.5): Connection refused
> nc: can't connect to remote host (192.168.0.5): Connection refused
> 2021-03-26T00:00:59-0700 Resetting wlan... eth1

**Comment by PomDev2** [comment]
Thanks. There is an issue with the quick reset method, and 0.7.6 has disabled it. See [this post](https://forums.slimdevices.com/showthread.php?111663-Community-Build-Radio-Firmware&p=1016345&viewfull=1#post1016345). Suggest installing that. 

However,  looking at your log files (thank you!), in particular wlanerr.log, it shows 
line 13: 
2021-03-26T00:00:27-0700 Squeezebox Bath.87_0752 failed 2021-03-26T00:00:14-0700 quick 2021-03-26T00:00:21-0700 reset 2021-03-25T23:38:11-0700 up 2021-03-26T00:00:21-0700 ...  fails 4 Ping 2s3q6f Fails[9]: 35326 137 1 0 396 11 4 0 473 62 0
The quick reset was used, the last full reset was 22 minutes earlier, the network recovered after 1 additional failed ping (fails 4), and you have had 473 quick and 62 full resets. Wow!

lines 15-25 are the next failure. line 25:
2021-03-26T00:00:45-0700 Squeezebox Bath.87_0752 failed 2021-03-26T00:00:32-0700 quick 2021-03-26T00:00:39-0700 reset 2021-03-25T23:38:11-0700 up 2021-03-26T00:00:39-0700 ... fails 4 Ping 2s3q6f Fails[9]: 35326 137 1 0 397 11 4 0 474 62 0
same as above, recovered a bit before 00:00:45 after 13 seconds outage. Your command line (via SSH) probably timed out or had some other problem. 

lines 27-37 yet another ping failure recovered by the quick reset, line 37:
2021-03-26T00:01:05-0700 Squeezebox Bath.87_0752 failed 2021-03-26T00:00:52-0700 quick 2021-03-26T00:00:59-0700 reset 2021-03-25T23:38:11-0700 up 2021-03-26T00:00:59-0700 ... fails 4 Ping 2s3q6f Fails[9]: 35327 137 1 0 398 11 4 0 475 62 0
Another 13 second restoration. But your ?ssh? session never received the recovery.

line 39-49:
2021-03-26T00:01:26-0700 Squeezebox Bath.87_0752 failed 2021-03-26T00:01:13-0700 quick 2021-03-26T00:01:20-0700 reset 2021-03-25T23:38:11-0700 up 2021-03-26T00:01:20-0700 ... fails 4 Ping 2s3q6f Fails[9]: 35327 137 1 0 399 11 4 0 476 62 0
And another 13 second restoration.

line 51:
2021-03-26T00:01:29-0700 Squeezebox Bath.87_0752 192.168.0.1  ping failed ...
and that's it until wlanpoke restarted:
2021-03-26T08:28:07-0700 Squeezebox Bath._0752 wlanpoke 749 at uptime: 08:28:07 up 0 min, load average: 1.06, 0.24, 0.07
without an ip

This would suggest that the connection continued to work, but that is impossible to believe, given 476 quick and 62 full resets up until that point. In 0.7.5,  additional last resort code was added to try to renew the dhcp lease, and log failures. The conditiona after a quick reset would rarely (but not rarely enough) trigger an entry to this "deactivation" state, but there are no logging entries to support that that happened here.

Still, it is important to try the script with the quick reset disabled. 0.7.6, now in the main branch, does that by default. You can also do this by adding "-Q 7" to your launch command in /etc/init.d/rcS.local, although do try to validate 0.7.6 if you can.

This is yet another mystery. I had a failure last night in which the log files showed nothing, but the SqueezePlay UI locked up hard when attempting to connect to another wireless network, requiring removing the battery to reboot. I remember something like this in the far distant past before the script, and possibly before the "troubles," but the details escape me. Problems with the quick reset method point to problems with the wpa_cli and wpa_supplicant daemons (demons?). 

Please run again with quick reset disabled. It will be interesting to see how many total failure you experience in a day compared with using the quick reset method. They may end up being fewer.  Thanks for your work!

**Comment by RichardDavies** [comment]
My network connection has remained stable since upgrading to v0.7.6. It's been up for 2 days and 6 hours at this point and reports:

> Ping 2s7q6f Fails[9]: 65498 246 0 0 0 0 0 0 0 130 0

Interestingly, it switched IP addresses, presumably during one of it's recoveries. It through me for a loop for a while because I couldn't access the quick webserver report page or connect via SSH, but everything on the radio was still working okay. Then I realized it'd been reassigned a new IP address by my router's DHCP server.

**Comment by PomDev2** [comment]
Good to know everything is fine with 0.7.6, A few more days, then on to 0.8.x?

You can add your radio's mac address to your router's static IP address reservations, so it doesn't change. Same for the LMS machine, and your desktop, etc.

I am suffering here with a quiet environment as they hinder testing improvements:
```
12:06 2021-03-29 ~2.5 days
      number of failed pings: ------------------------- ---- quick, full resets
dev/loc      params    index:     0    1   2  3 4 5 6 7 8  9 10 
Ofc  2S Ping 2s3q6f Fails[9]: 59948  281 100 18 3 4 4 0 0 27  0
bsmt  M Ping 2s3q6f Fails[9]: 79237  133   9  5 0 4 3 0 0  2  1
LR   1F Ping 2s3q6f Fails[9]: 78520  105   8  4 2 2 1 0 0 21  0
Kit  1E Ping 2s3q6f Fails[9]: 77519  215   5  7 0 2 0 0 0  5  3
MBR  2E Ping 2s3q6f Fails[9]: 69122 1425  49  6 0 0 0 0 0 23  3
Ofc  2N Ping 2s3q6f Fails[9]: 78511  347   0  0 0 0 0 0 0  1  3
bsmt  W Ping 2s3q6f Fails[9]: 75898  117   8  4 3 3 0 0 0  6  0
        ---- activity -------    ok       !q        f   q  f  wpa_cli restarts
```
Interesting you had no wpa_cli restarts. Some questions:
What version of the Squeezeplay software are you running? 
What are your signal levels? (Low signal levels delay dhcp and cause havoc given fixed full reset retry times.)
Have you enabled the 'slow' web server? (Soon to be the default.) Using gs.bat or gs.sh?

Based upon your 0.7.5.2 run's last entry:
`
2021-03-26T00:01:26-0700 Squeezebox Bath.87_0752 ... fails 4 Ping 2s3q6f Fails[9]: 35327 137 1 0 399 11 4 0 476 62 0
`
The 35327 trials at ~2.7 secs/trial average (w/o restarts) corresponds to a run of ~26 hours, implying that, assuming your environment was unchanged, the 0.7.5.2 quick reset method would have resulted in ~1000 quick restarts and about the same full restarts. If so, in this case, the 0.7.5.2 method was ineffective in lowering the full restarts. This is an important observation, thank you.

**Comment by RichardDavies** [comment]
> What version of the Squeezeplay software are you running?

Logitech Media Server Version: 7.7.6 running on a old Netgear ReadyNAS Duo

> What are your signal levels? (Low signal levels delay dhcp and cause havoc given fixed full reset retry times.)

Wireless Signal Strength: 100%

> Have you enabled the 'slow' web server? (Soon to be the default.) Using gs.bat or gs.sh?

No, I believe I've only tried using the quick web server so far.

Yeah, I'll let it run for another day or two and then upgrade to 0.8.x. Assuming the connection remains stable, I can also start upgrading my other remaining SB Radios that are still running 0.6.3. And yes, I know I can give it a static IP... I'm fine with it being dynamic---I just found it interesting that it happened to change while the device was still on and running.
